pkg/injector/bootstrap.go

```diff
@@ -100,3 +105,103 @@ func (wh *mutatingWebhook) marshalAndSaveBootstrap(name, namespace string, confi
 	log.Debug().Msgf("Creating bootstrap config for Envoy: name=%s, namespace=%s", name, namespace)
 	return wh.kubeClient.CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
 }
+
+// NewBootstrapSecretRotator returns a new bootstrap secret rotator.
+func NewBootstrapSecretRotator(ctx context.Context, kubeClient kubernetes.Interface, informerCollection *informers.InformerCollection, certManager *certificate.Manager, checkInterval time.Duration) *BootstrapSecretRotator {
+	return &BootstrapSecretRotator{
+		context:            ctx,
+		kubeClient:         kubeClient,
+		informerCollection: informerCollection,
+		certManager:        certManager,
+		checkInterval:      checkInterval,
+	}
+}
+
+// listBootstrapSecrets returns the bootstrap secrets stored in the informerCollection's store.
+// this function is used to get the namespace of the secrets.
+func (b *BootstrapSecretRotator) listBootstrapSecrets() []*corev1.Secret {
+	// informers return slice of pointers so we'll convert them to value types before returning
+	secretPtrs := b.informerCollection.List(informers.InformerKeySecret)
+	var secrets []*corev1.Secret
+
+	for _, secretPtr := range secretPtrs {
+		if secretPtr == nil {
+			continue
+		}
+		secret, ok := secretPtr.(*corev1.Secret)
+		if !ok {
+			continue
+		}
+		// finds bootstrap secrets
+		if strings.Contains(secret.Name, bootstrapSecretPrefix) {
```

Comment:
let's check the annotations here as well, there should be an annotation specific to OSM. 

app.kubernetes.io/instance=<meshName>
app.kubernetes.io/name=openservicemesh.io

Reply:
I believe this label filtering is already done in `informer.go`.